### PR TITLE
Adding implementaion of multiple runtime properties

### PR DESCRIPTION
**Pull request by lmanohara**
-Changed implementation to add multiple properties to a application.

### CODE REVIEW

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 1]
```diff
@@ -487,22 +513,24 @@ public void updateRuntimeProperties(List<RuntimeProperty> runtimeProperties, Dep
 
         }
 
-        Pod pod = kubernetesClient.pods().inNamespace(namespace.getMetadata().getName())
-                .withName(applicationContext.getId()).get();
+        PodList podList = kubernetesClient.pods().inNamespace(namespace.getMetadata().getName())
```
Comment (kasundsilva) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: Use getPods in KubernetesProvisioningUtils here

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesPovisioningConstants.java [hunk 2]
```diff
@@ -43,5 +43,5 @@
     public static final String HOST = "host";
 
     public static final String VOLUME_MOUNT = "default";
-    public static final String VOLUME_MOUNT_PATH = "/etc/default";
+    public static final String VOLUME_MOUNT_PATH = "/etc/";
```
Comment (danushkaf) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesPovisioningConstants.java]: Shouldn't we consider systems other than linux.

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 3]
```diff
@@ -184,17 +184,21 @@ public void archiveOrganization(TenantInfo tenantInfo) throws RuntimeProvisionin
             kubClient = KubernetesProvisioningUtils.getFabric8KubernetesClient();
             DeploymentList deploymentList = kubClient.extensions().deployments().list();
 
-            if (deploymentList.getItems().contains(deployment)) {
+            Deployment currentDeployement = kubClient.inNamespace(namespace.getMetadata().getName()).extensions()
```
Comment (danushkaf) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: for readability of the line shouldnt we consider seperating the line into several parts?

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 4]
```diff
@@ -184,17 +184,21 @@ public void archiveOrganization(TenantInfo tenantInfo) throws RuntimeProvisionin
...
                 kubClient.inNamespace(namespace.getMetadata().getName()).extensions()
                         .deployments().withName(config.getDeploymentName()).replace(deployment);
+
+                //todo need to change service replacement because need to check whether service object is available
                 //Service recreation should happen after comparing the new service config with running service configs.
-                for (Service service : serviceList) {
-                    kubClient.inNamespace(namespace.getMetadata().getName()).services().replace(service);
-                    serviceNameList.add(service.getMetadata().getName());
-                }
+//                for (Service service : serviceList) {
```
Comment (danushkaf) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: Dont keep commented code. Remove code if not used. If you need again you can find it from your commits.

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 5]
```diff
@@ -345,45 +368,44 @@ public void setRuntimeProperties(List<RuntimeProperty> runtimeProperties, Deploy
...
+                        .withMountPath(KubernetesPovisioningConstants.VOLUME_MOUNT_PATH + runtimeProperty.getName())
+                        .withReadOnly(true)
+                        .build();
+
+                volumeMounts.add(volumeMount);
+
                 break;
             case ENVIRONMENT:
                 if (log.isDebugEnabled()) {
                     String message = "Creating property type environment for the application:"
```
Comment (punnadi) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: add a space after colon

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesPovisioningConstants.java (hunk 2 above)
Comment (lmanohara) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesPovisioningConstants.java]: This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/c707570879201754759203ff99b9c5ea5489dcba

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 3 above)
Comment (lmanohara) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/c707570879201754759203ff99b9c5ea5489dcba

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 4 above)
Comment (lmanohara) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: This is code comment is removed from following commit.
https://github.com/wso2/product-af/pull/800/commits/c707570879201754759203ff99b9c5ea5489dcba

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 5 above)
Comment (lmanohara) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/c707570879201754759203ff99b9c5ea5489dcba

**Pull request merged**